src/union.ts

```diff
@@ -330,18 +331,121 @@ export class Union {
      * @param fs the filesystem interface to be added to the queue of FS's
      * @returns this instance of a unionFS
      */
-    use(fs: IFS): this {
-        this.fss.push(fs);
+    use(fs: IFS, options: VolOptions = {}): this {
+        this.fss.push([fs, options, this.createMethods(fs, options)]);
         return this;
     }
 
+    /**
+     * At the time of the [[use]] call, we create our sync, async and promise methods
+     * for performance reasons
+     *
+     * @param fs
+     * @param options
+     */
+    private createMethods(fs: IFS, options: VolOptions): FSMethodStack {
+        const noop = undefined
```

Comment:
While this works fine, it's throwing "undefined is not a function" which could be confusing if it ever gets seen by a user/developer, especially since this library is often used for mocking fs.

I think it would be better to throw a more explicit error, like "call to <method> failed as fs is not <writable/readable>".

Reply:
i'm a little confused here - do you have an example usage that would throw such an error?

The use of undefined here just shortcircuit the `[async|sync|promise]Method()` calls

Reply:
@G-Rath

Reply:
Lets circle back to this once you're refactored the core implementation logic, as it'll be much easier to properly understand whats going on :)

(I suspect you're right about this remaining `undefined` - what I'm wanting to avoid is calls being eaten and looking like they were successful, but that might require a bit of extra logic)